### PR TITLE
chore: add CODEOWNERS and repo config

### PR DESCRIPTION
**Pull request by JustinBeckwith**

### CODE REVIEW

**Review by hkdevandla — APPROVED** [review]

File: .github/sync-repo-settings.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  isAdminEnforced: true
+  requiredStatusCheckContexts:
+    - 'ci/circleci: build'
+    - 'ci/circleci: protobufjs-load-test'
+    - 'ci/circleci: push-generated-source'
+    - 'ci/circleci: test'
+    - 'ci/circleci: typescript-smoke-test'
+    - 'cla/google'
+    - 'codecov/patch'
+    - 'codecov/project'
```
Comment (noahdietz) [inline .github/sync-repo-settings.yaml:15]: Hi, can we please not include these are required checks? There is some weird flakey bug that drops our coverage on an uncoverable line which makes codecov unhappy. Maybe there is a codecov config thing I could change, but idk.

**Review by noahdietz — CHANGES_REQUESTED** [review]
Please see comment on PR WRT codecov

File: .github/sync-repo-settings.yaml (hunk 1 above)
Comment (JustinBeckwith) [reply .github/sync-repo-settings.yaml]: Ironically, I wouldn't have been able to merge this PR 😆 Fixed!

Comment (noahdietz) [reply .github/sync-repo-settings.yaml]: Thank you!!

**Review by noahdietz — APPROVED** [review]

**Pull request merged**